lib/Sema/TypeCheckInvertible.cpp

```diff
@@ -159,17 +159,28 @@ static void tryEmitContainmentFixits(InFlightDiagnostic &&diag,
 /// MARK: conformance queries
 
 static bool conformsToInvertible(CanType type, InvertibleProtocolKind ip) {
-  assert(!type->hasTypeParameter() && "forgot to mapTypeIntoContext first");
   auto &ctx = type->getASTContext();
 
+  auto *invertible = ctx.getProtocol(getKnownProtocolKind(ip));
+  assert(invertible);
+
   // Pack expansions such as `repeat T` themselves do not have conformances,
   // so check its pattern type for conformance.
   if (auto *pet = type->getAs<PackExpansionType>()) {
     type = pet->getPatternType()->getCanonicalType();
   }
 
-  auto *invertible = ctx.getProtocol(getKnownProtocolKind(ip));
-  assert(invertible);
+  // Must either not have a type parameter, or in the case of a
+  // BoundGenericXType, have a nominal available.
```

Comment:
I'm not sure I understand the comment. A BoundGenericType is not the only type that can both conform to protocols and contain type parameters, and the generic environment of the BGT's declaration is not necessarily the correct one for interpreting the BGT's generic argument types.